src/api/app/services/gitlab_status_reporter.rb

```diff
@@ -0,0 +1,28 @@
+class GitlabStatusReporter < SCMStatusReporter
+  def call
+    gitlab_client = Gitlab.client(endpoint: "#{@event_subscription_payload[:api_endpoint]}/api/v4",
+                                  private_token: @scm_token)
+    # https://docs.gitlab.com/ce/api/commits.html#post-the-build-status-to-a-commit
+    gitlab_client.update_commit_status(@event_subscription_payload[:project_id],
+                                       @event_subscription_payload[:commit_sha],
+                                       @state,
+                                       status_options)
+  rescue Gitlab::Error::Error => e
+    rescue_with_handler(e) || raise(e)
+  end
```

Comment:
We need to send metrics to RabbitMq after this rescue. Let's wait until PR #12799  is merged because it changes that code a bit.